### DISCUSSION TITLE
Camera API (Android): denying permissions for Photos/Video does nothing and still works!

### DISCUSSION THREAD

**Discussion by braincomb** [ionic-team/capacitor#6367] [General] [answered]
I'm seriously puzzled by this, all of a sudden denying permissions to Photos and Video gallery doesn't do anything, and the gallery still opens, no errors thrown! I can even see the permissions returned as 'denied' but gallery still opens:

`permissions: {"camera":"prompt","photos":"denied"}`

Using:
@capacitor/android@4.4.0
@capacitor/app@4.1.0
@capacitor/camera@4.1.3

Tested on S8 (Android 9) and S23 (Android 13) with the same results!

Here's a sample code pretty much directly from their demo app:

```ts
  const pickImagesFromGallery = async () => {
    try {
      const images = await Camera.pickImages({
        quality: 80,
        width: 1024,
      });

      for (const image of images.photos) {
        const fileName = new Date().getTime() + ".jpeg";
        const savedFileImage = await savePicture(image, fileName);
        photos.value = [savedFileImage, ...photos.value];
      }
    } catch (e) {
      // This should throw an error if user denied permissions but we never get here
      console.log(e);
      }
    }
  };
```

The other Camera API stuff with taking  camera pictures (`getPhoto()`) works as expected and throws an error if user denied permissions, I'd expect the same for `pickImages()`.

Before submitting a bug I wanted to see if anyone has been experiencing this or am I just tripping?

Edit:

I see from this Android documentation:

> Your app might allow users to choose from their photos and videos, such as for message attachments or profile pictures.
> 
> To support this functionality, use the photo picker. The photo picker doesn't require any runtime permissions to use. When a user interacts with the photo picker to select photos or videos to share with your app, the system grants temporary read access to the URI associated with the selected media files.

So it seems like runtime permissions are being granted for `pickImages`, so I just want to understand if this is expected behavior?  Although it seems like a confusing behavior for the end-user.

Additionally, if all I use is `pickImages`, does it mean I no longer need to declare READ_EXTERNAL_STORAGE and WRITE_EXTERNAL_STORAGE in the Manifest.xml?

**Comment by jcesarmobile** [comment, marked as answer]
The permission is needed for some devices and/or some gallery apps, it's impossible to know which devices or apps, so we request the permission but still proceed even if rejected, and in case it's one of those devices that require the permission and getting the image fails we return a permission error.
